pkg/apiserver/storage/file/file.go

```diff
@@ -482,35 +458,46 @@ func (s *Storage) GetList(ctx context.Context, key string, opts storage.ListOpti
 		return err
 	}
 
-	for _, obj := range objs {
-		currentVersion, err := s.versioner.ObjectResourceVersion(obj)
-		if err != nil {
-			return err
-		}
+	if v.IsNil() {
+		v.Set(reflect.MakeSlice(v.Type(), 0, 0))
+	}
 
-		if opts.SendInitialEvents == nil || (opts.SendInitialEvents != nil && !*opts.SendInitialEvents) {
-			// Apply the minimum resource version validation when we are not being called as part of Watch
-			// SendInitialEvents flow
-			// reason: the resource version of currently returned init items will always be < list RV
-			// they are being generated for, unless of course, the requestedRV == "0"/""
-			if err := s.validateMinimumResourceVersion(opts.ResourceVersion, currentVersion); err != nil {
-				// Below log left for debug. It's usually not an error condition
-				// klog.Infof("failed to assert minimum resource version constraint against list version")
+	remainingItems := (*int64)(nil)
+	for i, obj := range objs {
+		if opts.ResourceVersionMatch == v1.ResourceVersionMatchExact {
+			currentVersion, err := s.versioner.ObjectResourceVersion(obj)
+			if err != nil {
+				return err
+			}
+			expectedRV, err := s.versioner.ParseResourceVersion(opts.ResourceVersion)
+			if err != nil {
+				return err
+			}
+			if currentVersion != expectedRV {
 				continue
 			}
 		}
 
+		//if opts.SendInitialEvents == nil || (opts.SendInitialEvents != nil && !*opts.SendInitialEvents) {
```

Comment:
??? comment out or remove?

Reply:
removed in https://github.com/grafana/grafana/pull/90151/commits/7d173efe465bffe194460deea2ae65a75c142126